KimYeSeung/210930/Scripts/playermove.cs

```diff
@@ -0,0 +1,312 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+
+
+public class playermove : MonoBehaviour
+{
+      public Gamemanager gameManager;
+      public canscript canScript;
+      public BGM bgm;
+      public sevenHP Seven;
+
+      public AudioClip DioJump;
+      public AudioClip DioMove;
+
+      AudioSource audioSource;
+
+
+      public float maxSpeed;
+      public float jumpPower;
+      public float SecJumpPower;
+      public float JumpHap;
+      public float NoDamageTime;
+      public int JumpCount=2;
+      public int check = 1;
+
+
+
+      Rigidbody2D rigid;
+      SpriteRenderer spriteRenderer;
+      Animator anim;
+      BoxCollider2D boxcollider;
+      int Q = 0;
+
+      float timecheck = 0;
+
+
+
+      void Awake() {
+      rigid = GetComponent<Rigidbody2D>();
+      spriteRenderer = GetComponent<SpriteRenderer>();
+      anim = GetComponent<Animator>();
+      boxcollider = GetComponent<BoxCollider2D>();
+      this.audioSource=GetComponent<AudioSource>();
+
+      }
+
+      void PlaySound(string action) {
+            switch(action){
+            case "JUMP":
+                  audioSource.clip = DioJump;
+                  break;
+            
+            case "MOVE":
+                  audioSource.clip = DioMove;
+                  break;
+            }
+            audioSource.Play();
+
+      }
+
+      // void PlayBGM(string action){
+      //       switch(action){
+      //       case "0":
+      //             audioSource.clip = DioJump;
+      //             break;
+            
+      //       case "1":
+      //             audioSource.clip = DioMove;
+      //             break;
+      //       }
+      //       audioSource.Play();
+
+      // }
+
+      private void Update() {
+            float h = Input.GetAxisRaw("Horizontal");
+
+            if(JumpCount < 0){
+                  JumpCount = 0;
+            }
+
+            if(rigid.velocity.y < -20){
+                  timecheck += 1;
+            }
+
+
+            //점프
+            if(JumpCount > 0){
+                  if(Input.GetButtonDown("Jump")){
+                        PlaySound("JUMP");
+                        if(JumpCount == 1){
+                              if(rigid.velocity.y < 0){
+                                    rigid.AddForce(Vector2.up * JumpHap, ForceMode2D.Impulse);
+                              }
+                              rigid.AddForce(Vector2.up * SecJumpPower, ForceMode2D.Impulse);
+                              anim.SetBool("jump", true);
+                        }
+                        else{
+                              rigid.AddForce(Vector2.up * jumpPower, ForceMode2D.Impulse);
+                              anim.SetBool("jump", true);
+                        }
+                        --JumpCount;
+                  }
+            }
+
+            //방향전환
+            if(Input.GetButton("Horizontal")){
```

Comment:
```suggestion
            var axis = Input.GetAxisRaw("Horizontal");
            if(axis != 0)
            {
                spriteRenderer.flipX = axis < 0
            }
```
이러면 Input 관련 메소드를 두번씩이나 쓸 필요가 없고 axis 라는 값 하나로 스프라이트를 뒤집을지까지 결정할 수 있습니다.